sdk/eventhub/Azure.Messaging.EventHubs/src/Processor/InMemoryPartitionManager.cs

```diff
@@ -0,0 +1,155 @@
+    public class InMemoryPartitionManager : PartitionManager
+    {
+        /// <summary>The primitive for synchronizing access during ownership claim.</summary>
+        private readonly object OwnershipClaimLock = new object();
+
+        /// <summary>The set of stored checkpoints.  Partition ids are used as keys.</summary>
+        private ConcurrentDictionary<string, Checkpoint> Checkpoints;
+
+        /// <summary>The set of stored ownership.  Partition ids are used as keys.</summary>
+        private ConcurrentDictionary<string, PartitionOwnership> Ownership;
+
+        /// <summary>Logs activities performed by this partition manager.</summary>
+        private Action<string> Logger;
+
+        /// <summary>
+        ///   Initializes a new instance of the <see cref="InMemoryPartitionManager"/> class.
+        /// </summary>
+        ///
+        /// <param name="logger">Logs activities performed by this partition manager.</param>
+        ///
+        public InMemoryPartitionManager(Action<string> logger = null)
+        {
+            Logger = logger;
+
+            Checkpoints = new ConcurrentDictionary<string, Checkpoint>();
+            Ownership = new ConcurrentDictionary<string, PartitionOwnership>();
+        }
+
+        /// <summary>
+        ///   Retrieves a complete ownership list from the in-memory storage service.
+        /// </summary>
+        ///
+        /// <param name="eventHubName">The name of the specific Event Hub the ownership are associated with, relative to the Event Hubs namespace that contains it.</param>
+        /// <param name="consumerGroup">The name of the consumer group the ownership are associated with.</param>
+        ///
+        /// <returns>An enumerable containing all the existing ownership for the associated Event Hub and consumer group.</returns>
+        ///
+        public override Task<IEnumerable<PartitionOwnership>> ListOwnershipAsync(string eventHubName,
+                                                                                 string consumerGroup)
+        {
+            return Task.Run(() =>
+                Ownership
+                    .Select(kvp => kvp.Value)
+                    .Where(ownership => ownership.EventHubName == eventHubName &&
+                        ownership.ConsumerGroup == consumerGroup)
+            );
+        }
+
+        /// <summary>
+        ///   Tries to claim a list of specified ownership.
+        /// </summary>
+        ///
+        /// <param name="partitionOwnership">An enumerable containing all the ownership to claim.</param>
+        ///
+        /// <returns>An enumerable containing the successfully claimed ownership.</returns>
+        ///
+        public override Task<IEnumerable<PartitionOwnership>> ClaimOwnershipAsync(IEnumerable<PartitionOwnership> partitionOwnership)
+        {
+            return Task.Run(() =>
+                {
+                    List<PartitionOwnership> claimedOwnership = new List<PartitionOwnership>();
```

Comment:
Likewise, I'd suggest `Task.FromResult` here since there's no external communication.   Same for the other areas of the class that don't have an actual background process or external communication taking place.